src/main/java/org/spongepowered/api/tag/Tag.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.tag;
+
+import org.spongepowered.api.ResourceKeyed;
+import org.spongepowered.api.registry.DefaultedRegistryValue;
+
+import java.util.Collection;
+
+/**
+ * Represents a Tag for a given type.
+ * Tags group {@link TagTypes}.
+ *
+ * @param <T> Type that this tag groups
+ */
+public interface Tag<T> extends DefaultedRegistryValue, ResourceKeyed {
+
+    /**
+     * Get every value in this Tag.
+     *
+     * @return All tag values
+     */
+    Collection<T> all();
```

Comment:
You're adding `values` in the template, so should this be `values` to be consistent?